supervisord.conf

```diff
@@ -26,6 +26,17 @@ stdout_logfile_maxbytes=0
 stderr_logfile=/dev/stderr
 stderr_logfile_maxbytes=0
 
+[program:send_email_worker]
```

Comment:
@PiotrPrzybylak can you elaborate why are we going against single process per container design?

Reply:
Do you suggest we should have separate container per each conductor task worker?

Reply:
According to docker docs it is not recommended to have one container being responsible for multiple aspects of your overall application, but it is ok to have multiple processes in one container.
In the docs there is Apache web server, which starts multiple worker processes, given as an example.
So having multiple worker processes in conductor-workers container is ok for me.

Reply:
Apache is forking internal processes, it's still executing one application in container.  
We seem to have intention of running multiple individual workers.

Consider case: one of workers crashes/dies, this forces container restart as it should. At the same we have another worker half way though operation which get shut down, with container exit.

What does `autorestart` option do?

Reply:
how come it's only coming up now...?

to my mind that was always meant to be a single, generic container for N workers, although we discussed all of this in panic mode just before your vacation.

I wouldn't want container-per-worker simply because it's a waste. our setup is slow and cumbersome as it is. if you want this to be split, of course we can estimate and do just that.

back to your question - autorestart restarts the container when supervisord dies. supervisord dies if any of the workers dies. if the worker is halfway through - it never returns to conductor, meaning the task is still available on the queue. so it will be picked up again. so this is where the question of idempotency comes in.

if the tasks has side effects - they'll might happen twice. unless we also introduce some fake-idempotency (like - recognizing request ids and not doing anything if we've seen one before).

Reply:
Was looking at something else and noticed that there is shared container.

We need to split it up eater way as the second worker supposed to go into closed repository - there is bug opened for that. Each worker is pretty much microservice listening for events on the queue, which does not sound that different form any other microservice we have.

Containers are as expensive as processes they run, there is no substantial difference in cost between one container with two processes vs two containers.